source/s3_client.c

```diff
@@ -60,9 +60,15 @@ const uint32_t g_num_conns_per_vip_meta_request_look_up[AWS_S3_META_REQUEST_TYPE
 /* Should be max of s_num_conns_per_vip_meta_request_look_up */
 const uint32_t g_max_num_connections_per_vip = 10;
 
-/* TODO Provide more information on these values. */
+/**
+ * Default part size is 8 MB to reach the best performance from the experiments we had.
+ * Default max part size it SIZE_MAX, which is around 4GB.
```

Comment:
SIZE_MAX is 4GB on 32bit builds
but it's waaaaaaaaaaaaaay bigger on 64bit builds

like this?
`static const uint64_t s_default_max_part_size = SIZE_MAX < 5000000000000ULL ? SIZE_MAX : 5000000000000ULL;